### PR TITLE
Add more types support for recsys challenge cast operation

### PR DESCRIPTION
**Pull request by Elena-Qiu**
add more types: boolean, short
add corresponding check
remove pyspark.sql.types from documentation

### CODE REVIEW

File: pyzoo/test/zoo/friesian/feature/test_table.py [hunk 1]
```diff
@@ -453,6 +453,11 @@ def test_cast(self):
         assert dict(tbl.df.dtypes)['name'] == dict(tbl.df.dtypes)['a'] == dict(tbl.df.dtypes)['b'] \
             == dict(tbl.df.dtypes)['c'] == "float", \
             "all the columns should now be cast to float type"
+        with self.assertRaises(Exception) as context:
+            tbl = tbl.cast("a", "bool")
```
Comment (hkvision) [inline pyzoo/test/zoo/friesian/feature/test_table.py:457]: shall we support bool? i.e. do some mapping, bool&boolean; str&string; int&integer?

Comment (Elena-Qiu) [reply pyzoo/test/zoo/friesian/feature/test_table.py:457]: Ah, okay

Comment (Elena-Qiu) [reply pyzoo/test/zoo/friesian/feature/test_table.py:457]: so we can have any of {bool, boolean, str, string, int, integer}?

Comment (hkvision) [reply pyzoo/test/zoo/friesian/feature/test_table.py:457]: what do you think? is it reasonable for these mappings?

Comment (Elena-Qiu) [reply pyzoo/test/zoo/friesian/feature/test_table.py:457]: I think it's reasonable. I will add it.

File: pyzoo/zoo/friesian/feature/table.py [hunk 2]
```diff
@@ -300,10 +300,10 @@ def cast(self, columns, type):
...
         Cast columns to the specified type.
 
-        :param columns: a string or a list of strings that specifies column names. If it is None,
-                        then cast all of the columns.
-        :param type: a string ("string", "int", "long", "float", "double")
-                     or one of pyspark.sql.types that specifies the type.
+        :param columns: a string or a list of strings that specifies column names.
+                        If it is None, then cast all of the columns.
+        :param type: a string ("str/string", "bool/boolean", "int/integer", "long", "short",
+                               "float", "double") that specifies the type.
```
Comment (hkvision) [inline pyzoo/zoo/friesian/feature/table.py:306]: We just do the mapping in our code. In the document, we just need to tell users one option, i.e. string, boolean, int

File: pyzoo/zoo/friesian/feature/table.py [hunk 3]
```diff
@@ -312,8 +312,15 @@ def cast(self, columns, type):
...
-            raise ValueError("type should be a string or a dataype in pyspark.sql.types")
+        valid_types = ["str", "string", "bool", "boolean", "int",
+                       "integer", "long", "short", "float", "double"]
+        if (not isinstance(type, str) or (type not in valid_types)) \
+           and not isinstance(type, DataType):
+            raise ValueError(
+                "type should be a string in (str/string, bool/boolean, int/integer, "
+                "long, short, float, double).")
+        valid_dict = {"str": "string", "bool": "boolean", "integer": "int"}
+        type = valid_dict[type] if type in valid_dict.keys() else type
```
Comment (hkvision) [inline pyzoo/zoo/friesian/feature/table.py:323]: just type in valid_dict?

File: pyzoo/zoo/friesian/feature/table.py [hunk 4]
```diff
@@ -312,8 +312,15 @@ def cast(self, columns, type):
         elif not isinstance(columns, list):
             columns = [columns]
             check_col_exists(self.df, columns)
-        if not isinstance(type, str) and not isinstance(type, DataType):
-            raise ValueError("type should be a string or a dataype in pyspark.sql.types")
+        valid_types = ["str", "string", "bool", "boolean", "int",
+                       "integer", "long", "short", "float", "double"]
+        if (not isinstance(type, str) or (type not in valid_types)) \
```
Comment (hkvision) [inline pyzoo/zoo/friesian/feature/table.py:317]: Change to `not (isinstance(type, str) and type in valid_types)`?

File: pyzoo/zoo/friesian/feature/table.py [hunk 5]
```diff
@@ -312,8 +312,15 @@ def cast(self, columns, type):
...
+           and not isinstance(type, DataType):
+            raise ValueError(
+                "type should be a string in (str/string, bool/boolean, int/integer, "
```
Comment (hkvision) [inline pyzoo/zoo/friesian/feature/table.py:320]: type should be string, boolean, int, ...

File: pyzoo/zoo/friesian/feature/table.py (hunk 3 above)
Comment (Elena-Qiu) [reply pyzoo/zoo/friesian/feature/table.py:323]: No. It will transform, e.g. "str" to "string", otherwise it will keep the original value.

Comment (hkvision) [reply pyzoo/zoo/friesian/feature/table.py]: I mean change to valid_dict.keys() to valid_dict directly...

Comment (Elena-Qiu) [reply pyzoo/zoo/friesian/feature/table.py]: Sorry for big misunderstanding! Fixed now.

**Comment by Elena-Qiu** [comment]
Jenkins link: http://10.239.176.111:18888/view/ZOO-PR/job/ZOO-PR-Validation/5713/

**Pull request merged**